src/Stratis.SmartContracts.CLR.Tests/SmartContracts/SystemContractsDictionary.cs

```diff
@@ -0,0 +1,205 @@
+        for (int i = 0; i < signatories.Length; i++)
+        {
+            if (signatories[i] == address)
+            {
+                found = true;
+                for (int j = i + 1; j < signatories.Length; j++)
+                    signatories[j - 1] = signatories[j];
+
+                System.Array.Resize(ref signatories, signatories.Length - 1);
+            }
+        }
+
+        Assert(found, "The signatory does not exist.");
+        Assert(newSize == signatories.Length, "The expected size is incorrect.");
+
+        uint nonce = this.State.GetUInt32($"GroupNonce:{group}");
+
+        this.VerifySignatures(signatures, $"{nameof(RemoveSignatory)}(Nonce:{nonce},Group:{group},Address:{address},NewSize:{newSize},NewQuorum:{newQuorum})");
+
+        this.State.SetArray($"Signatories:{group}", signatories);
+        this.State.SetUInt32($"Quorum:{group}", newQuorum);
+        this.State.SetUInt32($"GroupNonce:{group}", nonce + 1);
+    }
+
+    public bool IsWhiteListed(UInt256 codeHash)
+    {
+        Assert(codeHash != default(UInt256));
+
+        WhiteListEntry whiteListEntry = this.State.GetStruct<WhiteListEntry>(codeHash.ToString());
+
+        return whiteListEntry.CodeHash != default(UInt256);
+    }
+
+    public UInt256 GetCodeHash(string name)
+    {
+        Assert(!string.IsNullOrEmpty(name));
+
+        return this.State.GetUInt256($"ByName:{name}");
+    }
+
+    public Address GetContractAddress(string name)
+    {
+        Assert(!string.IsNullOrEmpty(name));
+
+        UInt256 codeHash = this.State.GetUInt256($"ByName:{name}");
+
+        if (codeHash == default(UInt256))
+            return default(Address);
+
+        WhiteListEntry whiteListEntry = this.State.GetStruct<WhiteListEntry>(codeHash.ToString());
+
+        return whiteListEntry.LastAddress;
+    }
+
+    public Address GetContractAddress(UInt256 codeHash)
+    {
+        Assert(codeHash != default(UInt256));
+
+        WhiteListEntry whiteListEntry = this.State.GetStruct<WhiteListEntry>(codeHash.ToString());
+
+        return whiteListEntry.LastAddress;
+    }
+
+    public void WhiteList(byte[] signatures, UInt256 codeHash, Address lastAddress, string name)
+    {
+        Assert(signatures != null);
+        Assert(codeHash != default(UInt256));
+
+        UInt256 codeHashKey = codeHash;
+        WhiteListEntry whiteListEntry;
+
+        if (!string.IsNullOrEmpty(name))
+        {
+            codeHashKey = this.State.GetUInt256($"ByName:{name}");
+
+            if (codeHashKey == default(UInt256))
+                codeHashKey = codeHash;
+        }
+
+        whiteListEntry = this.State.GetStruct<WhiteListEntry>(codeHashKey.ToString());
+
+        uint nonce = this.State.GetUInt32($"Nonce:{codeHash}");
+
+        string authorizationChallenge;
+        if (whiteListEntry.CodeHash == default(UInt256))
+        {
+            authorizationChallenge = $"{nameof(WhiteList)}(Nonce:{nonce},CodeHash:{codeHash},LastAddress:{lastAddress},Name:{name})";
```

Comment:
Why not as below ? 

authorizationChallenge = $"{nameof(WhiteList)}:{nonce}:{codeHash},{lastAddress},{name})"

Reply:
It's meant to be human readable to ensure signatories can it read it and know what they are signing.